### PR TITLE
Time service

### PR DESCRIPTION
**Pull request by willsb**
:clock1: Service is useful for:
 
:clock2: Abstracting time 
:clock3: Help testing 
:clock4: Synchronize UI that depends on the current time

etc  :clock5:, etc :clock6:, etc :clock7: 

Squash if there’s no time to wait :clock8: (there actually is no time at all :clock9:)

### CODE REVIEW

**Comment by willsb** [comment]
Good call! This whole interface basically screams for us to expose `IConnectableObservable` instead of a simple `IObservable` to make its intentions clear. Here, have some: @paul 🌮  🌮

File: Toggl.Foundation/TimeService.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        public TimeService(IScheduler scheduler)
+        {
+            CurrentDateTimeObservable =
+                Observable
+                    .Interval(TimeSpan.FromSeconds(1), scheduler)
+                    .Select(_ => DateTimeOffset.UtcNow)
+                    .Publish();
+
+            disposable = CurrentDateTimeObservable.Connect();
```
Comment (amulware) [inline Toggl.Foundation/TimeService.cs]: Do we have to hold on to this?

Comment (willsb) [inline Toggl.Foundation/TimeService.cs]: I don't see harm in doing so

File: Toggl.Daneel/Startup/Setup.cs [hunk 2]
```diff
@@ -1,68 +1,72 @@
...
-            return navigationService;
+        {
+            navigationService = base.InitializeNavigationService();
+            return navigationService;
         }
-
+
         protected override IMvxApplication CreateApp()
         {
             base.InitializeFirstChance();
```
Comment (arturs105) [inline Toggl.Daneel/Startup/Setup.cs]: I know this isn't a change in this PR, but why is this called from `CreateApp()`?

File: Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs [hunk 3]
```diff
@@ -9,16 +9,19 @@ namespace Toggl.Foundation.MvvmCross.ViewModels
 {
```
Comment (arturs105) [inline Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs]: GH doesn't allow adding comments to the first 8 lines, but the first `using` statement has become unnecessary.

File: Toggl.Daneel/Startup/Setup.cs (hunk 2 above)
Comment (willsb) [inline Toggl.Daneel/Startup/Setup.cs]: Because it's responsible for creating the `IMvxApplication`. Part of the MvvmCross lifecycle

**Pull request merged**